### PR TITLE
linux: handle case where ZBOOT EFI stub is the target

### PR DESCRIPTION
**Pull request by jmbaur**
## Description of changes

When CONFIG_EFI_ZBOOT is enabled in the linux kernel's config, `vmlinuz.efi` becomes the file you want to install through the `zinstall` make target. This also needs the hexdump tool during the build of the EFI image.

## Things done

- Built on platform(s)
  - [x] x86_64-linux (cross to aarch64)
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by alyssais — APPROVED** [review]
Diff LGTM

**Comment by jmbaur** [comment]
Sorry for the churn, forgot this was targeting staging and had rebased on master

**Comment by jmbaur** [comment]
@alyssais any chance you'd be able to merge this one? I'm pretty sure amjoseph used to maintain the linux derivations, so there's no maintainer here to merge this anymore.

**Comment by JohnRTitor** [comment]
Hello, could you rebase this?

**Review by JohnRTitor — APPROVED** [review]
Ok lets merge this one after CI passes green

**Comment by Ma27** [comment]
@jmbaur 

> I'm pretty sure amjoseph used to maintain the linux derivations, so there's no maintainer here to merge this anymore.

What makes you think that? There were even people pinged by ofborg because there's a maintainer team for the kernel.

The sole problem - in my case - is that I just didn't have the time so far to take a look.

**Comment by jmbaur** [comment]
@Ma27 sorry the history looks kinda weird! I accidentally rebased this PR on master, when it was targeted at staging, so it pulled in a few unrelated files which then pulled in the maintainers for those files. IIRC, the only original reviewer that got pulled in was amjoseph.

File: pkgs/os-specific/linux/kernel/manual-config.nix
Comment (fpletz) [inline pkgs/os-specific/linux/kernel/manual-config.nix:124]: I tried building `vmlinuz.efi` on aarch64. This is actually wrong here since it needs to be put in `nativeBuildInputs`.

**Pull request merged**
